### ISSUE TITLE
cudf dataframe not working with DMatrix creation

### ISSUE THREAD

**Issue by kenhester** [rapidsai/cudf#484]
## Feature request
cudf dataframe not working with DMatrix creation

## Reporting a bug
Using the latest ngc container from 2 days ago, creating a DMatrix using a cudf dataframe still is not working or is not performant.  Results and times container in this issue.
System: DGX-1v 
Container: nvcr.io/nvidia/rapidsai/rapidsai:cuda9.2_ubuntu16.04 

```
rows_train = int(int(len(gdf_train)) * PERCENT_TRAIN)
print(rows_train)
rows_train = int(int(len(gdf_train)) * PERCENT_TRAIN)
print(rows_train)
```
```
36672493
```
```
%%time 
gx_train = gx[:rows_train]
gx_test = gx[rows_train:]
gy_train = gy[:rows_train]
gy_test = gy[rows_train:]
```
```
CPU times: user 404 ms, sys: 520 ms, total: 924 ms
Wall time: 923 ms
```
```
%%time
pgx_train = gx_train.to_pandas()
pgy_train = gy_train.to_pandas()
pgx_test = gx_test.to_pandas()
pgy_test = gy_test.to_pandas()
```
```
CPU times: user 58.5 s, sys: 5.37 s, total: 1min 3s
Wall time: 4.36 s
```
```
%%time
#pandas
dtrain_test = xgb.DMatrix(pgx_train.values, label=pgy_train.values, nthread=-1)
dtest_test = xgb.DMatrix(pgx_test.values, nthread=-1)
```
```
CPU times: user 38.7 s, sys: 10.3 s, total: 49 s
Wall time: 20.8 s
```
```
%%time
#cudf 
dtrain = xgb.DMatrix(gx_train, label=gy_train)
dtest = xgb.DMatrix(gx_test)
```
```
CPU times: user 52min 4s, sys: 1min 57s, total: 54min 2s
Wall time: 54min 1s
```

- [x ] I am using the latest version of cuDF from conda, built from `master`, or
built from the latest tagged release. 
- [x ] I have included the version or commit hash I am using for reference.
- [ ] I have included the following environment details:
      Linux Distro, Linux Kernel, GPU Model
- [ ] I have included the following version information for:
      Arrow, cmake, CUDA, gcc/g++, Numpy, Pandas, Python
- [ ] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

**Comment by datametrician** [comment]
@kenhester this works if you use dask in the container, that said for single GPU use it should be available as well.  We will add this as a feature to cuDF.

**Comment by datametrician** [comment]
That said, most systems are multi-gpu, and we strongly encourage you to use dask to benefit from all the GPU.

**Comment by randerzander** [comment]
This should be resolved in the 0.6 release container, which includes a build of [our xgboost fork with cudf-interop](https://github.com/rapidsai/xgboost) for single GPU work, and also our fork of [dask-xgboost](https://github.com/rapidsai/dask-xgboost) built to work with dask-cudf for multi-gpu support.

Closing for now. @kenhester please re-open if you try the above and it's not working for your use-case.
